redbadger/website-honestly
# People Listing CMS

### Motivation

Allows Badgers to be maintained in Prismic.
The following outlines the changes in this and the badger brain Pull Requests.

## Prismic
* Created a new Badger type in Prismic on Staging. The field details are taken fro the [Badger Profile Fields doc](https://docs.google.com/document/d/1NllUgMelb2XQ0yAmB88xCrvzowBGwfu4HT61R0D1E4I/edit?ts=585a8e5c)
* Used the tags field in Prismic to capture the categories for filtering the Badgers.

## Badger Brain
* This PR goes with the PR on badger brain, https://github.com/redbadger/badger-brain/pull/87
* All the Badger fields on Prismic are available on the GraphQL Api even though the website doesn't need them all

## Website Honestly
* The list of categories isn't hard-coded. Instead it's created from the categories assigned to the Badgers. This means there won't be any blank categories in the UI. It also means that any category a Badger is tagged with will appear in the UI so Prismic editors must be careful when tagging Badgers.
* The list of categories deduced from the Badgers returned is added to site state

### Test plan

Unit test the Badger type in badger brain
Unit test the site compile in website honestly
There's no UI so no browser testing required

#### To test on staging:
Check you can create/update/delete badgers in Prismic, following these [guidelines](http://webdocs.red-badger.com/updating-badgers.html)
These badgers won’t appear in the website because the website UI ticket hasn’t been done.
But you can test that these badgers are available through GraphQL:
* Log into Prismic on staging
* Check you can access all badgers from Prismic. Paste the following query into the left hand side:
```
{
  allBadgers {
    id
    firstName
    lastName
    jobTitle
    startDate
    imageUrl
    about
    skills
    influence
    achievements
    linkedin
    twitter
    github    
    categories {
      name
      slug
    }
  }
}
```
* Click the play button and check that the results are returned on the right
* Check you can access a badger by Id from Prismic. Paste the following query into the left hand side, replacing the id with one of the ids returned from the previous query for all badgers:
```
{
  badger (id: "someid") {
    firstName
    lastName
    jobTitle
    startDate
    imageUrl
    about
    skills
    influence
    achievements
    linkedin
    twitter
    github    
    categories {
      name
      slug
    }
  } 
}
```
* Click the play button and check that the right badger is returned on the right

### Pre-merge checklist
